### PR TITLE
document OPENSHIFT-ADMIN-OUTPUT-RULES

### PR DESCRIPTION
**Pull request by danwinship**
Document the new iptables chain that exists for admin special purposes.
https://github.com/openshift/origin/pull/14221 / https://trello.com/c/dnZHuI5R
@openshift/networking

### CODE REVIEW

**Review by knobunc — APPROVED** [review]
Nice clean docs.  Thanks.  I flagged it enterprise-3.6 since I don't think we are back-porting this, right?

**Comment by danwinship** [comment]
isn't 3.6 still in the future and "backporting" would be 3.5?
but we haven't backported the upstream code yet anyway...

**Comment by knobunc** [comment]
@danwinship -- right.  I just wanted to make sure the docs end up in the correct release for where the feature is going.  3.6 has not been released and is what we are working on at the moment.

File: admin_guide/managing_networking.adoc [hunk 1]
```diff
@@ -435,6 +440,44 @@ $ oc create -f <replication_controller>.json
...
+
+For example, you could create rules that log traffic to particular
+destinations, or to prevent more than a certain number of outgoing
+connections per second.
+
+{product-title} does not provide a way to add custom iptables rules
+automatically, but it does provide a place where such rules can be
+added manually by the administrator. Each node, on startup, will
+create an empty chain called `OPENSHIFT-ADMIN-OUTPUT-RULES` in the
+`filter` table (assuming that that chain doesn't already exist). Any
```
Comment (pravisankar) [inline admin_guide/managing_networking.adoc]: assuming that chain?

**Review by pravisankar — APPROVED** [review]
LGTM

File: admin_guide/iptables.adoc [hunk 2]
```diff
@@ -32,6 +32,13 @@ not impact the operation of {product-title} and the Docker service. Also, change
 will often need to be made on all nodes in the cluster. Use caution, as iptables
 is not designed to have multiple concurrent users and it is very easy to break
 {product-title} and Docker networking.
+
+{product-title} provides one chain which is specifically intended for
```
Comment (ahardin-rh) [inline admin_guide/iptables.adoc]: chain, which

File: admin_guide/managing_networking.adoc [hunk 3]
```diff
@@ -435,6 +440,44 @@ $ oc create -f <replication_controller>.json
 oc describe rc <replication_controller>
 ----
 
+[[admin-guide-limit-pod-access-iptables]]
+=== Using iptables rules to Limit Access to External Resources
```
Comment (ahardin-rh) [inline admin_guide/managing_networking.adoc]: Rules

File: admin_guide/managing_networking.adoc [hunk 4]
```diff
@@ -435,6 +440,44 @@ $ oc create -f <replication_controller>.json
...
+
+Some cluster administrators may want to perform actions on outgoing
+traffic that don't fit within the model of EgressNetworkPolicy or the
```
Comment (ahardin-rh) [inline admin_guide/managing_networking.adoc]: s/don't/do not

File: admin_guide/managing_networking.adoc (hunk 1 above)
Comment (ahardin-rh) [inline admin_guide/managing_networking.adoc]: s/doesn't/does not

File: admin_guide/managing_networking.adoc (hunk 4 above)
Comment (ahardin-rh) [inline admin_guide/managing_networking.adoc]: `EgressNetworkPolicy`

File: admin_guide/managing_networking.adoc [hunk 5]
```diff
@@ -435,6 +440,44 @@ $ oc create -f <replication_controller>.json
...
+There are a few things to watch out for when using this functionality:
+
+. It is up to you to ensure that rules get created on each node;
+{product-title} does not provide any way to make that happen
+automatically.
+
+. The rules are not applied to traffic that exits the cluster via an
+egress-router, and they run after EgressNetworkPolicies are applied
+(and so will not see traffic that is denied by an
+EgressNetworkPolicy).
```
Comment (ahardin-rh) [inline admin_guide/managing_networking.adoc]: EgressNetworkPolicy

File: admin_guide/managing_networking.adoc [hunk 6]
```diff
@@ -435,6 +440,44 @@ $ oc create -f <replication_controller>.json
...
+to any other traffic).
+
+There are a few things to watch out for when using this functionality:
+
+. It is up to you to ensure that rules get created on each node;
+{product-title} does not provide any way to make that happen
+automatically.
+
+. The rules are not applied to traffic that exits the cluster via an
+egress-router, and they run after EgressNetworkPolicies are applied
```
Comment (ahardin-rh) [inline admin_guide/managing_networking.adoc]: egress router

File: admin_guide/iptables.adoc (hunk 2 above)
Comment (danwinship) [inline admin_guide/iptables.adoc]: Adding a comma implies (to me) that openshift only provides one chain (and that chain is intended for administrators to use). But it creates many chains, it's just that only one of them has this purposes.

**Comment by danwinship** [comment]
github thinks this is still running in travis, but travis says it passed...

File: admin_guide/iptables.adoc (hunk 2 above)
Comment (bfallonf) [inline admin_guide/iptables.adoc]: @danwinship @ahardin-rh I'd agree with Ashley. Commas are really only for separating clauses. I'd suggest "OpenShift provides a number of chains, one of which is specifically intended for..."

I was asked to have a look at this PR and see if it needed a followup and this is the only thought I had. I'll do that.

**Pull request merged**